src/main/java/uk/gov/hmcts/reform/prl/services/ManageOrderService.java

```diff
@@ -2035,4 +2038,25 @@ public void setMarkedToServeEmailNotification(CaseData caseData, Map<String, Obj
             caseDataUpdated.put("markedToServeEmailNotification", No);
         }
     }
+
+    public YesOrNo getAllChildrenFinalOrderIssuedStatus(CaseData caseData) {
+
+        YesOrNo finalOrderStatus = null;
+        List<Child> children = caseData
+            .getChildren()
+            .stream()
+            .map(Element::getValue)
+            .collect(Collectors.toList());
+
+        List<YesOrNo> finalOrderIssuedList = children.stream()
+            .map(Child::getIsFinalOrderIssued)
+            .collect(Collectors.toList());
+        log.info("before setting the isFinalOrderIssuedForAllChildren flag ==={}===", finalOrderStatus);
+
+        for (YesOrNo isFinalOrder: finalOrderIssuedList) {
+            finalOrderStatus = Yes.equals(isFinalOrder) ? Yes : No;
```

Comment:
say you have 3 children in the case. order is closed for only 3rd and not the first 2.
This logic will return Yes saying it is final.
I meant, this line 2057 will always return the status of the last YesOrNo, it will not work I feel